### PR TITLE
CMA-19 Add messages functionality

### PR DESCRIPTION
**Pull request by weronikac**
# add endpoints for messages functionality

### CODE REVIEW

File: server/src/main/java/com/project/cisco/controler/MessageApiController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.project.cisco.api.MessageApi;
+import com.project.cisco.dto.MessageDto;
+import com.project.cisco.service.MessageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
```
Comment (BlackDilvish) [inline server/src/main/java/com/project/cisco/controler/MessageApiController.java:13]: Let's add here origin to fix an error releated to CORS on frontend

`@CrossOrigin(value = { "http://localhost:3000/" },
        maxAge = 900
)`

File: server/src/main/java/com/project/cisco/service/impl/MessageServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    private LanguageRepository languageRepository;
+    @Autowired
+    private TagRepository tagRepository;
+    @Autowired
+    private MessageRepository messageRepository;
+    @Autowired
+    private MessageMapper messageMapper;
+
+    @Override
+    public MessageDto addMessage(MessageDto messageDto) {
```
Comment (BlackDilvish) [inline server/src/main/java/com/project/cisco/service/impl/MessageServiceImpl.java:34]: We have a use case specifying that: "Translation should have same tags as original message". I believe that should be on a backend side, to check if `original_message` has any tags and copy them to the new one.

**Comment by AleksandraRolka** [comment]
@weronikac please add basic requests for messages functionalities in:
src/test/java/com/project/cisco/postman/**cisco-app.postman_collection.json**

**Comment by AleksandraRolka** [comment]
You should also take care of this case

image
because now it does not work like that, it throws exception:
image


please check also others requirements for messages if are fulfilled

**Comment by AleksandraRolka** [comment]
If on backend Deleting, Modifying English will be disabled you could remove this options in Operations column for that language

File: server/src/main/java/com/project/cisco/controler/MessageApiController.java (hunk 1 above)
Comment (weronikac) [reply server/src/main/java/com/project/cisco/controler/MessageApiController.java:13]: done

**Review by AleksandraRolka — APPROVED** [review]

**Pull request merged**